tests/vmi_multus_test.go

```diff
@@ -580,6 +580,36 @@ var _ = Describe("SRIOV", func() {
 
 	tests.BeforeAll(func() {
 		tests.BeforeTestCleanup()
+		// Check if the hardware supports SRIOV
+		nodes := tests.GetAllSchedulableNodes(virtClient)
+		Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
+
+		for _, node := range nodes.Items {
+			var sriovallocate bool = false
+			var sriovcapacity bool = false
+			resourceList := node.Status.Allocatable
+			for k, v := range resourceList {
+				if strings.HasPrefix(string(k), sriovResourceName) {
+					if v.Value() > 0 {
+						sriovallocate = true
+						break
+					}
+				}
+			}
+			capacityList := node.Status.Capacity
+			for k, v := range capacityList {
+				if strings.HasPrefix(string(k), sriovResourceName) {
+					if v.Value() > 0 {
+						sriovcapacity = true
+						break
+					}
+				}
+			}
+			if (!sriovallocate || !sriovcapacity){
+				Skip("SRIOV is not enabled, test skipped")
```

Comment:
I would vote to use an early `Fail` instead. SR-IOV is enabled in tests, it is just not available on the cluster. As a user of these tests I would like to be clearly notified (the skip can be easily lost in the output). On the other hand, I understand that it is frustrating to get fail from the test-suite if you don't have specific hardware.

Would you mind using `Fail` instead and explain in the message why it failed and how can users skip the suite?

This PR would be a suggestion to skip all tests without sufficient cluster environment. I think it is dangerous. Failing early and with clear explanation would be a better service.

Reply:
+1